benchmarks/src/commonMain/kotlin/BufferOps.kt

```diff
@@ -0,0 +1,331 @@
+                + "sell it."),
+        "2bytes" to "\u0080\u07ff",
+        "3bytes" to "\u0800\ud7ff\ue000\uffff",
+        "4bytes" to "\ud835\udeca",
+        // high surrogate, 'a', low surrogate, and 'a'
+        "bad" to "\ud800\u0061\udc00\u0061")
+
+    @Param("20", "2000", "200000")
+    var length = 0
+
+    @Param("ascii", "utf8", "sparse", "2bytes", "3bytes", "4bytes", "bad")
+    var encoding: String = "ascii"
+
+    private var string: String = ""
+
+    private fun constructString(): String {
+        val part = strings[encoding] ?: throw IllegalArgumentException("Unsupported encoding: $encoding")
+        val builder = StringBuilder(length + 1000)
+        while (builder.length < length) {
+            builder.append(part)
+        }
+        builder.setLength(length)
+        return builder.toString()
+    }
+
+    override fun padding(): ByteArray {
+        val baseString = constructString()
+        if (baseString.utf8Size() >= minGap) {
+            return baseString.encodeToByteArray()
+        }
+        val builder = StringBuilder((minGap * 1.5).toInt())
+        while (builder.length < minGap) {
+            builder.append(baseString)
+        }
+        return builder.toString().encodeToByteArray()
+    }
+
+    @Setup
+    fun setupString() {
+        string = constructString()
+    }
+
+    @Benchmark
+    fun benchmark(): String {
+        val s = buffer.size
+        buffer.writeUtf8(string)
+        return buffer.readUtf8(buffer.size - s)
+    }
+}
+
+open class Utf8LineBenchmarkBase: BufferRWBenchmarkBase() {
+    @Param("17")
+    var length: Int = 0
+
+    @Param("LF", "CRLF")
+    var separator: String = ""
+
+    protected var string: String = ""
+
+    private fun lineSeparator(): String = when (separator) {
+        "LF" -> "\n"
+        "CRLF" -> "\r\n"
+        else -> throw IllegalArgumentException("Unsupported line separator type: $separator")
+    }
+
+    private fun constructString(): String = ".".repeat(length) + lineSeparator()
+
+    override fun padding(): ByteArray {
+        val string = constructString()
+        if (string.length >= minGap) {
+            return string.encodeToByteArray()
+        }
+        val builder = StringBuilder((minGap * 1.5).toInt())
+        while (builder.length < minGap) {
+            builder.append(string)
+        }
+        return builder.toString().encodeToByteArray()
+    }
+
+    @Setup
+    fun setupString() {
+        string = constructString()
+    }
+}
+
+open class Utf8LineBenchmark: Utf8LineBenchmarkBase() {
+    @Benchmark
+    fun benchmark(): String? {
+        buffer.writeUtf8(string)
+        return buffer.readUtf8Line()
+    }
+}
+
+open class Utf8LineStrictBenchmark: Utf8LineBenchmarkBase() {
+    @Benchmark
+    fun benchmark(): String {
+        buffer.writeUtf8(string)
+        return buffer.readUtf8LineStrict()
+    }
+}
+
+private const val INDEX_OF_TARGET_VAL: Byte = 1
```

Comment:
Maybe call something like "Indexed Value"?
Actually, the byte value itself is stored here, and not the index by which it is located.

Reply:
The intent was to name it like "`indexOf`'s target value", but now I see it didn't work out. :)
Sure, I'll peek a different name.